test/java/seedu/addressbook/commands/ViewCommandTest.java

```diff
@@ -0,0 +1,161 @@
+        Command command = new ViewAllCommand(targetIndex);
+        command.setData(addressBook, relevantPersons);
+
+        return command;
+    }
+
+    /**
+     * Asserts that both a ViewCommand and a ViewAllCommand can retrieve from
+     * the {@code addressBook} details of the person at the given {@code index} in the given {@code list}.
+     *
+     * @param index one-indexed position of the target person in the list
+     */
+    private void assertViewSuccess(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
+        ReadOnlyPerson personToBeViewed = list.get(index - 1); // -1 is because index is one-indexed
+
+        String expectedMessage = String.format(ViewCommand.MESSAGE_VIEW_PERSON_DETAILS,
+                                               personToBeViewed.getAsTextHidePrivate());
+        assertViewCommandBehaviour(addressBook, list, index, expectedMessage);
+
+        expectedMessage = String.format(ViewCommand.MESSAGE_VIEW_PERSON_DETAILS,
+                                        personToBeViewed.getAsTextShowAll());
+        assertViewAllCommandBehaviour(addressBook, list, index, expectedMessage);
+    }
+
+    /**
+     * Asserts that the details of person at specific index cannot be retrieved due to
+     * invalid index.
+     */
+    private void assertViewErrorInvalidIndex(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
+        assertCommandError(addressBook, list, index, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Asserts that the details of person at specific index cannot be retrieved due to
+     * person not existing in the addressbook.
+     */
+    private void assertViewErrorPersonNotInAddressBook(AddressBook addressBook, List<ReadOnlyPerson> list, int index) {
+        assertCommandError(addressBook, list, index, Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
+    }
+
+    /**
+     * Asserts that the Viewcommand and ViewAllcommand reports the given error for the given input
+     */
+    private void assertCommandError(AddressBook addressBook, List<ReadOnlyPerson> list, int index,
+                                    String expectedMessage) {
+        assertViewCommandBehaviour(addressBook, list, index, expectedMessage);
+        assertViewAllCommandBehaviour(addressBook, list, index, expectedMessage);
+    }
+
+    /**
+     * Asserts that the View command gives given feedback information
+     */
+    private void assertViewCommandBehaviour(AddressBook addressBook, List<ReadOnlyPerson> list, int index, 
+                                            String expectedMessage) {
+        Command command = generateViewCommand(addressBook, list, index);
+        assertCommandResult(command, addressBook, TestUtil.clone(addressBook), expectedMessage);
+    }
+
+    /**
+     * Asserts that the ViewAll command gives given feedback information
+     */
+    private void assertViewAllCommandBehaviour(AddressBook addressBook, List<ReadOnlyPerson> list, int index,
```

Comment:
`assertViewCommandBehaviour()` and `assertViewCommandBehaviour()` look very similar, which hints they can be combined into a more generic method.

By the way, these methods are still using British spelling.